subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/matcher/TextMatchers.kt

```diff
@@ -0,0 +1,135 @@
+package com.avito.android.test.matcher
+
+import android.content.res.Resources
+import android.view.View
+import android.widget.TextView
+import androidx.test.espresso.matcher.BoundedDiagnosingMatcher
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.hamcrest.StringDescription
+
+internal object TextMatchers {
+
+    internal fun withText(resourceId: Int) =
+        WithCharSequenceMatcher(
+            resourceId,
+            WithCharSequenceMatcher.TextViewMethod.GET_TEXT,
+        )
+
+    internal fun withText(text: String) =
+        withText(Matchers.`is`(text))
+
+    internal fun withText(stringMatcher: Matcher<String>) =
+        WithTextMatcher(stringMatcher)
+}
+
+internal class WithTextMatcher(
```

Comment:
These matches look like workarounds for an Espresso bug.
Please, file an issue (if not) and add a link.

You can make a reproducible example in `ui-testing-core-app` in a separate branch.

Reply:
There is no Espresso bug or something.

The only reason we do this is to enhance our error reports. 
Currently, you can see something like "expected: nbsp string; was: nbsp string". The reason being is an nsbp and a regular space-symbol are displayed equally. Apparently, this message looks confusing, it takes time to find out the root cause. It is even a bigger problem when there is multiple nbsp in string, debug time grows up dramatically.
I like the way the AS does this, we can do it the same way.
![image](https://user-images.githubusercontent.com/1591583/130744779-cbb8e811-bfa3-4253-86bf-3a6d5e3d5f56.png)

Reply:
You described that output is not expected and is confusing. This is by definition a bug. At least, from our perspective.
Maybe, maintainers will refuse to change behavior and it's their responsibility and right.
I propose to file a request. I see it as our responsibility as clients of the library - to give feedback. You even prepared a fix.

Reply:
This output is expected, cause it doesn't conflict with any standard.
However, it does make sense, I'll give it a try on proposing changes to the androidx repo later.